frontend/src/container/MetricsApplication/MetricsPageQueries/MetricsPageQueriesFactory.ts

```diff
@@ -184,12 +184,140 @@ export const getQueryBuilderQuerieswithAdditionalItems = ({
 	],
 });
 
+export const getQueryBuilderQueriesTop = ({
+	metricName,
+	servicename,
+	tagFilterItems,
+	legend,
+	topLevelOperations,
+}: BuilderQueriesPropsTop): {
+	formulas: IMetricsBuilderFormula[];
+	queryBuilder: IMetricsBuilderQuery[];
+} => ({
+	formulas: [],
+	queryBuilder: [
+		{
+			aggregateOperator: 18,
+			disabled: false,
+			legend,
+			metricName,
+			name: 'A',
+			reduceTo: 1,
+			tagFilters: {
+				items: [
+					{
+						id: '',
+						key: 'service_name',
+						op: 'IN',
+						value: [`${servicename}`],
+					},
+					{
+						id: '',
+						key: 'operation',
+						op: 'MATCH',
+						value: topLevelOperations,
+					},
+					...tagFilterItems,
+				],
+				op: 'AND',
+			},
+		},
+	],
+});
+export const getQueryBuilderQueriesTopWithFormula = ({
+	servicename,
+	legend,
+	legendFormula,
+	disabled,
+	tagFilterItems,
+	topLevelOperations,
+	metricNameA,
+	metricNameB,
+	groupBy,
+	expression,
+	additionalItems,
+}: BuilderQueriesTopWithFormulaProps): {
+	formulas: IMetricsBuilderFormula[];
+	queryBuilder: IMetricsBuilderQuery[];
+} => ({
+	formulas: [
+		{
+			disabled: false,
+			expression,
+			name: 'F1',
+			legend: legendFormula,
+		},
+	],
+	queryBuilder: [
+		{
+			aggregateOperator: 18,
+			disabled,
+			groupBy,
+			legend,
+			metricName: metricNameA,
+			name: 'A',
+			reduceTo: 1,
+			tagFilters: {
+				items: [
+					{
+						id: '',
+						key: 'service_name',
+						op: 'IN',
+						value: [`${servicename}`],
+					},
+					{
+						id: '',
+						key: 'operation',
+						op: 'MATCH',
+						value: topLevelOperations,
+					},
+					additionalItems,
```

Comment:
I don't want to nitpick, but please pay attention to the reader's perception. `additionalItems` reads like a list that should be expanded, but it's just a simple object. You shouldn't name plural to that. Unless I go look at the `additionalItems` I would think this is a list and would be buggy not to expand.

Reply:
@srikanthccv Yeah agreed.. When I first refactored the code for External tabs, I thought I could generalize it for additional items if any, for future. At that time I thought I might have to add "Top Level Operations" etc.
But now that I have refactored, it doesn't make any sense.